modules/imgproc/src/stackblur.cpp

```diff
@@ -0,0 +1,1261 @@
+
+                stack[stackStart * widthLen + k] = *(srcPtr + k);
+                sumIn[k] += *(srcPtr + k);
+                sum[k] += sumIn[k];
+
+                sumOut[k] += stack[sp1 * widthLen + k];
+                sumIn[k] -= stack[sp1 * widthLen + k];
+            }
+
+            dstPtr += widthElem;
+            ++sp;
+            if (sp >= kernelSize) sp = 0;
+        }
+    }
+
+private:
+    const Mat &src;
+    Mat &dst;
+    int radius;
+    int CN;
+    int height;
+    int widthElem;
+    int hm;
+    float mulVal;
+    int mulValTab;
+    int shrValTab;
+};
+
+void stackBlur(InputArray _src, OutputArray _dst, Size ksize)
+{
+    CV_INSTRUMENT_REGION();
+    CV_Assert(!_src.empty());
+
+    CV_Assert( ksize.width  > 0 && ksize.width  % 2 == 1 &&
+               ksize.height > 0 && ksize.height % 2 == 1 );
+
+    int radiusH = ksize.height / 2;
+    int radiusW = ksize.width / 2;
+
+    int stype = _src.type(), sdepth = _src.depth();
+    Mat src = _src.getMat();
+
+    if (ksize.width == 1)
+    {
+        _src.copyTo(_dst);
+
+        if (ksize.height == 1)
+            return;
+    }
+    else
+    {
+        _dst.create( src.size(), stype);
+    }
+
+    Mat dst = _dst.getMat();
+    int numOfThreads = getNumThreads();
+    int widthElem = src.cols * src.channels();
+
+    if (dst.rows / numOfThreads < 3)
+        numOfThreads = std::max(1, dst.rows / 3);
+
+    if (sdepth == CV_8U)
+    {
+        if (ksize.width != 1)
+            parallel_for_(Range(0, src.rows), ParallelStackBlurRow<uchar, int>(src, dst, radiusW), numOfThreads);
+        if (ksize.height != 1)
+            parallel_for_(Range(0, widthElem), ParallelStackBlurColumn<uchar, int>(dst, dst, radiusH), numOfThreads);
+    }
+    else if (sdepth == CV_16S)
+    {
+        if (ksize.width != 1)
+            parallel_for_(Range(0, src.rows), ParallelStackBlurRow<short, int>(src, dst, radiusW), numOfThreads);
+        if (ksize.height != 1)
+            parallel_for_(Range(0, widthElem), ParallelStackBlurColumn<short, int>(dst, dst, radiusH), numOfThreads);
+    }
+    else if (sdepth == CV_16U)
+    {
+        if (ksize.width != 1)
+            parallel_for_(Range(0, src.rows), ParallelStackBlurRow<ushort, int>(src, dst, radiusW), numOfThreads);
+        if (ksize.height != 1)
+            parallel_for_(Range(0, widthElem), ParallelStackBlurColumn<ushort, int>(dst, dst, radiusH), numOfThreads);
+    }
+    else if (sdepth == CV_32F)
+    {
+        if (ksize.width != 1)
+            parallel_for_(Range(0, src.rows), ParallelStackBlurRow<float, float>(src, dst, radiusW), numOfThreads);
+        if (ksize.height != 1)
+            parallel_for_(Range(0, widthElem), ParallelStackBlurColumn<float, float>(dst, dst, radiusH), numOfThreads);
+    }
+    else
+        CV_Error_( CV_StsNotImplemented,
+                   ("Unsupported input format in StackBlur, the supported formats are: CV_8U, CV_16U, CV_16S and CV_32F."));
```

Comment:
`CV_Error_` use cv::format internally.
It is not needed here, so common `CV_Error` should be used instead.